### DISCUSSION TITLE
Mandate test functions in all commands of the CLI

### DISCUSSION THREAD

**Discussion by t8** [nestdotland/nest.land#214] [General]
I'd like to open a discussion here to allow for peoples' opinions on whether or not to require test functions on the CLI. This would mean that for each new command and all existing commands, a corresponding test would have to be made. This way, any additions or deletions to the commands would be able to be tested automatically before allowing a PR to merge.

What is your opinion on this?

**Comment by littledivy** [comment]
@tbaumer22 Command like `init` that need user input can have an alternate `initx` command that takes in that input from arguments. What do you think?

**Comment by t8** [comment]
@divy-work that would work. Perhaps we could use the same "init" function, where the information is just passed in two different ways from the different commands. In a sense, each function would be shared by two different commands.

OR we could consider parsing optional arguments in the original command itself. Either way would work.
